controllers/update_test.go

```diff
@@ -1123,40 +1157,53 @@ func mockSignature(time time.Time) *git2go.Signature {
 	}
 }
 
-func clone(repoURL, remoteName, branchName string) (*git2go.Repository, error) {
+func clone(repoURL, branchName string) (*git2go.Repository, error) {
 	dir, err := os.MkdirTemp("", "iac-clone-*")
 	if err != nil {
 		return nil, err
 	}
+	transportOptsURL := "http://" + randStringRunes(5)
+	managed.AddTransportOptions(transportOptsURL, managed.TransportOptions{
+		TargetURL: repoURL,
+	})
```

Comment:
Since transport options now accept context, `clone()` can accept a context and pass it to transport option.